src/titiler/extensions/titiler/extensions/render.py

```diff
@@ -0,0 +1,182 @@
+
+    params: RenderItem
+    links: List[Link]
+
+
+class RenderItemList(BaseModel):
+    """List of Render Items with links."""
+
+    renders: Dict[str, RenderItemWithLinks]
+    links: List[Link]
+
+
+@define
+class stacRenderExtension(FactoryExtension):
+    """Add /renders endpoint to a STAC TilerFactory."""
+
+    def register(self, factory: MultiBaseTilerFactory):
+        """Register endpoint to the tiler factory."""
+
+        def _prepare_query_string(render: Dict, src_path: str) -> str:
+            """Prepare render related query params.
+
+            Validates and filters query params that titiler can understand.
+            If titiler does not support parameter, it will be ignored.
+            """
+            # List of dependencies a `/tile` URL should validate
+            # Note: Those dependencies should only require Query() inputs
+            tile_dependencies = [
+                factory.layer_dependency,
+                factory.dataset_dependency,
+                # Image rendering Dependencies
+                factory.rescale_dependency,
+                factory.color_formula_dependency,
+                factory.colormap_dependency,
+                factory.render_dependency,
+            ]
+
+            query = {"url": src_path}
+            for dependency in tile_dependencies:
+                dep = get_dependant(path="", call=dependency)
+                if dep.query_params:
+                    # call the dependency with the query-parameters values
+                    query_values, _errors = request_params_to_args(
+                        dep.query_params, render
+                    )
+                    _ = dependency(**query_values)
+                    query.update(query_values)
+            return urlencode(
+                {key: value for key, value in query.items() if value is not None},
+                doseq=True,
+            )
+
+        def _prepare_render_item(
+            render_id: str, render: Dict, request: Request, src_path: str
+        ) -> Dict:
+            """Prepare single render item."""
+            query_string = _prepare_query_string(render, src_path)
+            links = [
+                {
+                    "href": factory.url_for(
+                        request,
+                        "Show STAC render",
+                        render_id=render_id,
+                    )
+                    + "?"
+                    + urlencode({"url": src_path}),
+                    "rel": "self",
+                    "type": "application/json",
+                    "title": f"{render_id} render item",
+                },
+                {
+                    "href": factory.url_for(
+                        request,
+                        "tilejson",
+                        tileMatrixSetId="{tileMatrixSetId}",
+                    )
+                    + "?"
+                    + query_string,
+                    "rel": "tilesets-map",
```

Comment:
why `tilesets-map` ?

Reply:
Well, I guess the `Link` type definition was based on OGC Tiles API, and I followed the link and browsed available `rel`s
`tilesets-map` had the most convincing description:
> The target IRI points to a resource that describes how to provide tile sets of the context resource in map format.

Anyways, I picked it because I was a bit lost in how to describe the relation and I am super open to alternatives.